src/PeerSelectState.h

```diff
@@ -26,6 +26,8 @@ class HtcpReplyData;
 class HttpRequest;
 class icp_common_t;
 class StoreEntry;
+class CachePeer;
+using DisappearingCachePeer = CbcPointer<CachePeer>;
```

Comment:
Please add `src/peering.h` for _forward_ declarations related to cache peers, "neighbors", and peer selection; move these two declarations there. Please declare KeptCachePeer there as well, similar to how src/helper/forward.h forward-declares ClientPointer. Include `peering.h` as needed instead of duplicating these declarations.

Eventually, we will probable move many files into `src/peering/` and have `src/peering/forward.h`, but I would not crate that directory in this PR.

Reply:
Added.